packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -350,7 +356,11 @@ export class KeybindingWidget extends ReactWidget {
     protected renderRow(item: KeybindingItem, index: number): React.ReactNode {
         const { command, keybinding } = item;
         // TODO get rid of array functions in event handlers
```

Comment:
it should be rewritten like react component then item can be kept in props.

Reply:
> it should be rewritten like react component then item can be kept in props.

I finally had a bit of spare time to look back into performing the refactoring (rewrite the rows as a react component) and fix the event handlers. I'm stuck at the moment trying to figure out the clean way to be able to call and register methods in the `keymaps-frontend-contribution` which are ultimately defined in the react component. 

I'm not sure what the best approach would be to clean the code, make sure event handlers are correct and are not re-created each time, so if you have a chance to help it'd be greatly appreciated, I'll keep investigating and trying on my end.

My changes so far are tracked in the following [branch](https://github.com/eclipse-theia/theia/compare/vf/keymaps-refactor).